### ISSUE TITLE
salt-cloud: Creating EC2 ebs volume from snapshot ignores volume type

### ISSUE THREAD

**Issue by rusvdw** [saltstack/salt#30539]
I've declared an EC2  EBS volume in a cloud profile that is to be created from an existing snapshot. I've also specified that I want it to be a GP2 (SSD) volume:

```
volumes:
    - { device: /dev/sdf, type: gp2, snapshot: snap-63380d10 } 
 ```
The volume is created, but it is always a 'standard' magnetic volume, not the gp2 SSD volume that was requested.

```
root@ops2:~# salt --versions
Salt Version:
           Salt: 2015.8.0-1233-g03262c4
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: Not Installed
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.4.0
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-74-generic
         system: Ubuntu 14.04 trusty
```

**Comment by jfindlay** [comment]
@rusvdw, thanks for the report.

**Comment by rallytime** [comment]
@rusvdw I think #30677 will help with this bug. Can you give that a try and let us know if that is fixed for you?

**Comment by rallytime** [comment]
Fixed by #30677.
